src/http/server/request_processor.cr

```diff
@@ -47,18 +50,22 @@ class HTTP::Server::RequestProcessor
         begin
           @handler.call(context)
         rescue ex
-          response.respond_with_status(:internal_server_error)
-          error.puts "Unhandled exception on HTTP::Handler"
-          ex.inspect_with_backtrace(error)
+          unless response.closed?
+            Log.error(exception: ex) { "Unhandled exception on HTTP::Handler" }
```

Comment:
How does this play with `HTTP::ErrorHandler`? In case the error handler triggers, there won't be a log entry, right?

Reply:
Right, `HTTP::ErrorHandler` swallows all the exceptions already. This is used when `HTTP::Server` is used without handlers or when something more severe happens.

Reply:
So if you only use `ErrorHandler` you get less than without it (with respect to error logging) and you need to include `LogHandler` to go back to the behaviour that was already provided by plain `RequestProcessor`. This seems a bit awkward.

Reply:
Well, that's currently explained in the `ErrorHandler` doc. Maybe we could remove the logging from `RequestProcessor` so it doesn't get less 😆 

The way I see it, the handler at `RequestProcessor` is there as last resort when no middleware are used. Would you combine `ErrorHandler` and `LogHandler` in the same class? Or remove it altogether and have always the same behavior at `RequestProcessor`?

Anyway, I didn't change that in this PR and these classes are getting as old as Crystal I think. Let's open a RFC or a separate PR to modify this stack.

Reply:
Can we remove `ErrorHandler`?

Reply:
What I did was sneak peek how rack solves this issue and do something similar. Effectively, `LogHandler` must be the first handler to work properly and it doesn't handle or logs any error. It just logs requests. The `ErrorHandler`, if used, will send appropriate responses to the client if possible, and also log the errors.

So now, if you only use `LogHandler` the requests are logged, and the errors are still handled by the `RequestProcessor` (but not sent to the client).

If you only use `ErrorHandler` the errors are sent to the client and logged, just like the `RequestProcessor` do, although we might want to make it more detailed in the future. Another reason I found to log errors from here is that I think it's more clear that the exception is logged _before_ the request as it happens when `LogHander` is used alone.